lib/internal/http2/core.js

```diff
@@ -1833,21 +1852,25 @@ class Http2Stream extends Duplex {
     session[kMaybeDestroy]();
     process.nextTick(emit, this, 'close', code);
     callback(err);
-  }
 
+  }
   // The Http2Stream can be destroyed if it has closed and if the readable
   // side has received the final chunk.
   [kMaybeDestroy](error, code = NGHTTP2_NO_ERROR) {
-    if (error == null) {
-      if (code === NGHTTP2_NO_ERROR &&
-          (!this._readableState.ended ||
-          !this._writableState.ended ||
-          this._writableState.pendingcb > 0 ||
-          !this.closed)) {
-        return;
-      }
+    if (error || code !== NGHTTP2_NO_ERROR) {
```

Comment:
I would move the `code !== NGHTTP2...` check to the next if. That way it is clearer that destroy is not called with error in that case. Or do I miss something here?

Reply:
The code is correct as it is. There can be an `error` (from net or tls) or the stream can return an error code - but that's not mapped to a JS `ERROR` yet.